optimism/src/preimage_oracle.rs

```diff
@@ -24,6 +24,57 @@ pub struct ReadWrite<R, W> {
 
 pub struct RW(pub ReadWrite<PipeReader, PipeWriter>);
 
+// Here, we implement `os_pipe::pipe` in a way that allows us to pass flags. In particular, we
+// don't pass the `CLOEXEC` flag, because we want these pipes to survive an exec, and we set
+// `DIRECT` to handle writes as single atomic operations (up to splitting at the buffer size).
+// This fixes the IPC hangs. This is bad, but the hang is worse.
+
+#[cfg(not(any(target_os = "ios", target_os = "macos", target_os = "haiku", windows)))]
+fn create_pipe() -> std::io::Result<(PipeReader, PipeWriter)> {
+    let mut fds: [libc::c_int; 2] = [0; 2];
+    let res = unsafe { libc::pipe2(fds.as_mut_ptr(), libc::O_DIRECT) };
```

Comment:
I cannot find what are the default flags for `os_pipe::pipe`. If you have anything, I'm interested in.

Reply:
https://docs.rs/crate/os_pipe/latest/source/src/unix.rs